backend/User/views.py

```diff
@@ -82,6 +87,51 @@ def delete(self, request, user_id):
 		user.delete()
 		return Response(status=status.HTTP_204_NO_CONTENT)
 
+class UserProfilePictureView(APIView):
+	@method_decorator(csrf_exempt)
+	def post(self, request, user_id):
+		try:
+			user = User.objects.get(pk=user_id)
+		except User.DoesNotExist:
+			return Response(status=status.HTTP_404_NOT_FOUND)
+
+		if 'file' not in request.FILES:
+			return Response(status=status.HTTP_400_BAD_REQUEST)
+		file = request.FILES['file']
+
+		try:
+			validate_image(file)
+		except ValidationError as e:
+			return Response({"message": e.messages[0]}, status=status.HTTP_400_BAD_REQUEST)
+
+		try:
+			profile_picture = ProfilePicture.objects.get(user=user)
+			profile_picture.picture = file
+		except ProfilePicture.DoesNotExist:
+			profile_picture = ProfilePicture(user=user, picture=file)
+		profile_picture.save()
+		return Response(status=status.HTTP_200_OK)
+
+	def get(self, request, user_id):
+		try:
+			user = User.objects.get(pk=user_id)
+			profile_picture = ProfilePicture.objects.filter(user=user).first()
+			if not profile_picture:
+				return Response(status=status.HTTP_404_NOT_FOUND)
+
+			# Get the actual file path from the ImageFieldFile instance
+			image_path = profile_picture.picture.path
+			with open(image_path, "rb") as image_file:
+				image_data = image_file.read()
+				encoded_image = base64.b64encode(image_data).decode('utf-8')
+				return Response({'image': encoded_image}, status=status.HTTP_200_OK)
+		except User.DoesNotExist:
```

Comment:
Ah yeah. This is actually a little bit weird of a route as we handle in the UI basically the fact that something is not found. Basically this should never get here without a user that is actually logged in an existing so I was being lazy in this regard. What we do in the next PR is actually return an empty string and OK because in 90% of the cases it means the user did not load their own avatar -> we use the default one and no need to return an actual error :D But I am open to concerns! Jsut felt like returning error messages might also give false positives kind of.